Packs/PaloAltoNetworks_IoT3rdParty/Scripts/SendAllPANWIoTDevicesToServiceNow/SendAllPANWIoTDevicesToServiceNow.py

```diff
@@ -0,0 +1,187 @@
+        "asset_type": "device",
+        "output_format": "ServiceNow",
+        "servicenow_map": sn_query_result,
+        "asset_list": device_list,
+        "using": PANW_IOT_INSTANCE
+    })
+    return upsert_devices_resp[0]["Contents"]
+
+
+def create_servicenow_record(insert_list):
+    """
+    To create Servicenow record
+    :param insert_list: device list that needs to insert into Servicenow table
+    """
+    for x in range(len(insert_list)):
+        device = insert_list[x]
+        fields = device['fields']
+        custom_fields = device['custom_fields']
+        demisto.executeCommand("servicenow-create-record", {
+            "table_name": SERVICENOW_TABLE_NAME,
+            "fields": fields,
+            "custom_fields": custom_fields,
+            "using": SERVICENOW_INSTANCE
+        })
+
+
+def update_servicenow_record(update_list):
+    """
+    To update existing device record in Servicenow
+    :param update_list: device list that needs to be updated
+    """
+    for x in range(len(update_list)):
+        device = update_list[x]
+        fields = device['fields']
+        custom_fields = device['custom_fields']
+        sys_id = device['sys_id']
+        demisto.executeCommand("servicenow-update-record", {
+            "table_name": SERVICENOW_TABLE_NAME,
+            "fields": fields,
+            "custom_fields": custom_fields,
+            "id": sys_id,
+            "using": SERVICENOW_INSTANCE
+        })
+
+
+def get_all_panw_iot_devices_and_send_to_servicenow():
+    """
+    To send all devices from PANW IoT cloud to Servicenow
+    :return: A summary message
+    """
+    run_time_count = 1
+    offset = 0
+    update_num = 0
+    insert_num = 0
+    total_update_num = 0
+    total_insert_num = 0
+    while True:
+        size = 0
+        device_list = get_devices_from_panw_iot_cloud(offset)
+        size = len(device_list)
+        deviceid_list = [device['deviceid'] for device in device_list]
+        query = "mac_addressIN" + ",".join(deviceid_list)
+        sn_query_result = query_servicenow_table(query)
+
+        if len(sn_query_result) > 0:
+            upsert_devices_result = get_servicenow_upsert_device_list(sn_query_result, device_list)
+            insert_count = upsert_devices_result['insert_count']
+            update_count = upsert_devices_result['update_count']
+            total_update_num += update_count
+            total_insert_num += insert_count
+            update_num = update_count
+            insert_num = insert_count
+            if insert_count == 0 and update_count == 0:
+                err_msg = (
+                    f'Error, Bulk Servicenow sync failed to get upsert '
+                    f'devices from PANW IoT cloud {upsert_devices_result}'
+                )
+                send_status_to_panw_iot_cloud("error", err_msg)
+                demisto.info(f'{PANW_IOT_INSTANCE} {err_msg}')
```

Comment:
better to use LOG as it is gathers all and write it in the same time when the script ends.